🐛 BUG: Failure to report Authentication Error

### Which Cloudflare product(s) does this pertain to?

D1

### What version(s) of the tool(s) are you using?

3.8.0 -> 3.10.1

### What version of Node are you using?

_No response_

### What operating system are you using?

Linux

### Describe the Bug

I was running this command in ci: `wrangler d1 migrations apply customer_db --env staging` and my script would just die. No logging output, nothing I could do to get any information.

After numerous failures debugging the issue, I added this command: `wrangler d1 backup list customer_db --env staging` and viola...
```
✘ [ERROR] A request to the Cloudflare API (/accounts/{MYACCTID}/d1/database/{MYDBID}/backup) failed.
  Authentication error [code: 10000]
```

I went and updated my API Token, which didn't have D1 edit permissions and magically it all worked. 

Wrangler has to display an error message when not authenticated / authorized for the given functionality.

### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

_No response_

--- a/.changeset/strange-eels-search.md
+++ b/.changeset/strange-eels-search.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+fix: suggest checking permissions on authentication error with API token set

--- a/packages/wrangler/src/__tests__/deploy.test.ts
+++ b/packages/wrangler/src/__tests__/deploy.test.ts
@@ -8324,6 +8324,9 @@ export default{
 		  Authentication error [code: 10000]
 
 
+		📎 It looks like you've set a custom API token with an environment variable.
+		Please ensure it has the correct permissions for this operation.
+
 		Getting User settings...
 		👋 You are logged in with an API Token, associated with the email user@example.com!
 		┌───────────────┬────────────┐

--- a/packages/wrangler/src/__tests__/deploy.test.ts
+++ b/packages/wrangler/src/__tests__/deploy.test.ts
@@ -8324,7 +8324,7 @@ export default{
 		  Authentication error [code: 10000]
 
 
-		📎 It looks like you've set a custom API token with an environment variable.
+		📎 It looks like you are authenticating Wrangler via a custom API token set in an environment variable.
 		Please ensure it has the correct permissions for this operation.
 
 		Getting User settings...

--- a/packages/wrangler/src/index.ts
+++ b/packages/wrangler/src/index.ts
@@ -1,6 +1,7 @@
 import module from "node:module";
 import os from "node:os";
 import TOML from "@iarna/toml";
+import chalk from "chalk";
 import { ProxyAgent, setGlobalDispatcher } from "undici";
 import makeCLI from "yargs";
 import { version as wranglerVersion } from "../package.json";
@@ -56,7 +57,13 @@ import {
 import { tailOptions, tailHandler } from "./tail";
 import { generateTypes } from "./type-generation";
 import { printWranglerBanner } from "./update-check";
-import { listScopes, login, logout, validateScopeKeys } from "./user";
+import {
+	getAuthFromEnv,
+	listScopes,
+	login,
+	logout,
+	validateScopeKeys,
+} from "./user";
 import { vectorize } from "./vectorize/index";
 import { whoami } from "./whoami";
 
@@ -724,6 +731,13 @@ export async function main(argv: string[]): Promise<void> {
 			await createCLIParser([...argv, "--help"]).parse();
 		} else if (isAuthenticationError(e)) {
 			logger.log(formatMessage(e));
+			const envAuth = getAuthFromEnv();
+			if (envAuth !== undefined && "apiToken" in envAuth) {
+				const message =
+					"📎 It looks like you've set a custom API token with an environment variable.\n" +
+					"Please ensure it has the correct permissions for this operation.\n";
+				logger.log(chalk.yellow(message));
+			}
 			await whoami();
 		} else if (e instanceof ParseError) {
 			e.notes.push({

--- a/packages/wrangler/src/index.ts
+++ b/packages/wrangler/src/index.ts
@@ -734,7 +734,7 @@ export async function main(argv: string[]): Promise<void> {
 			const envAuth = getAuthFromEnv();
 			if (envAuth !== undefined && "apiToken" in envAuth) {
 				const message =
-					"📎 It looks like you've set a custom API token with an environment variable.\n" +
+					"📎 It looks like you are authenticating Wrangler via a custom API token set in an environment variable.\n" +
 					"Please ensure it has the correct permissions for this operation.\n";
 				logger.log(chalk.yellow(message));
 			}

